src/deepmol/datasets/datasets.py

```diff
@@ -12,15 +14,6 @@ class Dataset(ABC):
     Subclasses need to implement their own methods based on this class.
     """
 
-    def __init__(self) -> None:
-        self._features2keep = None
-
-    def __len__(self) -> int:
```

Comment:
Shouldn't this method be abstract and mandatory for child classes to implement?

Reply:
Hum, I don't think it needs to be an abstract method. It is kinda useless (what should it return? the number of molecules? or another thing?). Maybe we can keep it but it will be the same for every child class (no?) so I think it does not need to be an abstract method. What do you think?

Reply:
added __len__ as abstract method of Dataset class